types/block.go

```diff
@@ -229,39 +121,13 @@ func (b *Block) fillDataAvailabilityHeader() {
 	shares := namespacedShares.RawShares()
 
 	// create the nmt wrapper to generate row and col commitments
-	squareSize := uint32(math.Sqrt(float64(len(shares))))
-	tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(squareSize))
-
-	// TODO(ismail): for better efficiency and a larger number shares
-	// we should switch to the rsmt2d.LeopardFF16 codec:
-	extendedDataSquare, err := rsmt2d.ComputeExtendedDataSquare(shares, rsmt2d.NewRSGF8Codec(), tree.Constructor)
+	squareSize := uint64(math.Sqrt(float64(len(shares))))
```

Comment:
```suggestion
	squareSize := uint64(math.Sqrt(float64(dataSharesLen)))
```

Reply:
the naming is confusing here, and we are clearly not explicit enough in what compute shares returns. `dataSharesLen` is the number of shares that actually have data in them, but to determine the square size, we need to take the square root of the total number of shares. So if the block is not completely full, then `dataSharesLen` will not equal `len(shares)`, which is what we need.

will add docs

Reply:
docs add in https://github.com/celestiaorg/celestia-core/pull/524/commits/e4dc361be28d4d1aacca7e13893a90c899c06716

part of the reason I think this is confusing because `ComputeShares` is computing the shares along with padding the shares. We should probably decouple those things to both make it more explicit, and give us more control over the size of the square.

Reply:
yeah, dataSharesLen != len(shares)

>  We should probably decouple those things to both make it more explicit, and give us more control over the size of the square.

Yeah either that or, in the case more control is not necessary, rename `ComputeShares` to sth that makes it more explicit.